packages/forms-web-app/src/dynamic-forms/validator/validator.js

```diff
@@ -1,49 +1,38 @@
-// const { body, validationResult } = require('express-validator');
-// const hasQuestionnaire = require('../definitions/questionnaire/hasQuestionnaire');
-// const questionnaire = hasQuestionnaire.questionnaire;
-// const formHelper = require('../lib/dynamicformhelper');
+const RequiredValidator = require('./requiredValidator');
+const ValidOptionValidator = require('./validOptionValidator');
+const hasJourney = require('../has-questionnaire/journey');
 
-// const validate = () => {
-//     return async (req, res, next) => {
-//         const { appealId, section, question } = req.params;
-//         var questionObj = formHelper.getQuestionBySectionAndName(questionnaire, section, question);
-//         const validations = [];
-//         switch (questionObj.validator?.type) {
-//             case "text":
-//                 validations.push(textValidations(questionObj))
-//                 break;
-//             case "boolean":
-//                 validations.push(booleanValidations(questionObj))
-//                 break;
-//             case "boolean-text":
-//                 validations.push(booleanValidations(questionObj))
-//                 validations.push(textValidations(questionObj))
-//                 break;
-//         }
-//         await Promise.all(validations.map(validation => validation.run(req)));
+const validate = () => {
+	return async (req, res, next) => {
+		const { section, question } = req.params;
+		var questionObj = hasJourney.getQuestionBySectionAndName(section, question);
+		const validations = [];
+		questionObj.validators.forEach((validator) => {
+			switch (validator.constructor) {
+				case RequiredValidator:
+					validations.push(new RequiredValidator().validate(questionObj));
+					break;
+				case ValidOptionValidator:
+					validations.push(new ValidOptionValidator().validate(questionObj));
+					break;
+				default:
+					throw new Error(`Validator of type ${validator.constructor} not implemented`);
+			}
+		});
 
-//        next();
-//     };
-// };
+		const results = [];
+		let overallResult = true;
+		for (const validation of validations) {
+			let validationResult = await validation.run(req);
```

Comment:
I am assuming this is what the next middleware will do.

Reply:
Possibly something like this will achieve what we want, reading up on some examples we should be able to check the presence of any errors how you have it there 👍 
let validationResult = await validation.run(req);
if (validationResult.errors.length > 0) {

				next();
}